euphonic/spectra.py

```diff
@@ -478,22 +487,19 @@ def __getitem__(self, item: slice) -> SC:
 
     def __getitem__(self, item: Union[int, slice]):  # noqa: F811
         new_metadata = deepcopy(self.metadata)
-        if 'labels' in new_metadata.keys():
-            labels = new_metadata.pop('labels')
-        else:
-            labels = None
+        line_metadata = new_metadata.pop('line_data',
+                                         [{} for _ in self._y_data])
         if isinstance(item, int):
-            if labels is not None:
-                new_metadata['label'] = labels[item]
+            new_metadata = {**new_metadata, **line_metadata[item]}
```

Comment:
I don't think this will break? Using {**x, **y}, any duplicate keys in y will overwrite the keys in x. Although maybe  `update` is clearer, I've changed this and added a new test for duplicate keys

Reply:
You're quite right. I didn't realise this behaviour was defined, but it is! https://docs.python.org/3/reference/expressions.html#dict

I was thinking of function arguments, which do raise an error if there is redundancy. Still, `update()` does make this clearer.